hack/scripts/minikube-install.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+#  2024 NVIDIA CORPORATION & AFFILIATES
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an AS IS BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+if [[ "${TRACE-0}" == "1" ]]; then
+    set -o xtrace
+fi
+
+CLUSTER_NAME="${CLUSTER_NAME:-"net-op-dev"}"
+MINIKUBE_BIN="${MINIKUBE_BIN:-"unknown"}"
+MINIKUBE_DRIVER="${MINIKUBE_DRIVER:-"unknown"}"
+USE_MINIKUBE_DOCKER="${USE_MINIKUBE_DOCKER:-"true"}"
+NUM_NODES="${NUM_NODES:-"1"}"
+NODE_MEMORY="${NODE_MEMORY:-"4g"}"
+NODE_CPUS="${NODE_CPUS:-"2"}"
+NODE_DISK="${NODE_DISK:-"20g"}"
+
+## Detect the OS.
+OS="unknown"
+if [[ "${OSTYPE}" == "linux"* ]]; then
+  OS="linux"
+elif [[ "${OSTYPE}" == "darwin"* ]]; then
+  OS="darwin"
+fi
+
+# Exit if the OS is not supported.
+if [[ "$OS" == "unknown" ]]; then
+  echo "os '$OSTYPE' not supported. Aborting." >&2
+  exit 1
+fi
+
+## Set the driver used for minikube machines. By default this script will select the preferred VM driver for each OS.
+## Users can override this using the MINIKUBE_DRIVER env variable.
+if [[ "$MINIKUBE_DRIVER" == "unknown" ]]; then
+  if [[ "${OSTYPE}" == "linux"* ]]; then
+    MINIKUBE_DRIVER="kvm2"
+  elif [[ "${OSTYPE}" == "darwin"* ]]; then
+    MINIKUBE_DRIVER="qemu"
+  fi
+fi
+
+MINIKUBE_ARGS="${MINIKUBE_ARGS:-"\
+  --driver $MINIKUBE_DRIVER \
+  --cpus=$NODE_CPUS \
+  --memory=$NODE_MEMORY \
+  --disk-size=$NODE_DISK \
+  --nodes=$NUM_NODES \
+  --preload=true \
+  --cni calico \
+  --container-runtime=docker\
+  --install-addons \
+  --addons registry"}"
+
+MINIKUBE_EXTRA_ARGS="${MINIKUBE_EXTRA_ARGS:-""}"
+
+echo "Setting up Minikube cluster..."
+
+## Exit early if the cluster already exists.
+if [[ $($MINIKUBE_BIN status -p $CLUSTER_NAME  -f '{{.Name}}') == $CLUSTER_NAME ]]; then
+  echo "Minikube cluster '$CLUSTER_NAME' found. Skipping cluster set-up"
+  ## Set environment variables to use the Minikube VM docker build for
+  if [[ "$USE_MINIKUBE_DOCKER" == "true" ]]; then
+    echo "Setting environment variables to use $CLUSTER_NAME as docker build environment."
+    eval $($MINIKUBE_BIN -p $CLUSTER_NAME docker-env)
+  fi
+  exit 0
+fi
+
+$MINIKUBE_BIN start --profile $CLUSTER_NAME $MINIKUBE_ARGS
+
+## Set environment variables to use the Minikube VM docker build for
+if [[ "$USE_MINIKUBE_DOCKER" == "true" ]]; then
+  echo "Setting environment variables to use $CLUSTER_NAME as docker build environment."
+  eval $($MINIKUBE_BIN -p $CLUSTER_NAME docker-env)
```

Comment:
This is not working correctly on my setup.
* Mac Silicon
* zsh as default shell

I think it's related to how this script is started. 

Effectively, I need to run this command manually otherwise I get:
```
$ make dev-skaffold
CLUSTER_NAME=net-op-dev MINIKUBE_BIN=/Users/vremmas/repos/network-operator/bin/minikube-v0.0.0-20231012212722-e25aeebc7846 /Users/vremmas/repos/network-operator/hack/scripts/minikube-install.sh
Setting up Minikube cluster...
Minikube cluster 'net-op-dev' found. Skipping cluster set-up
Setting environment variables to use net-op-dev as docker build environment.
kubectl apply -f hack/crds/*   ## Deploy the network attachment definition CRD.
customresourcedefinition.apiextensions.k8s.io/network-attachment-definitions.k8s.cni.cncf.io unchanged
/Users/vremmas/repos/network-operator/bin/skaffold-v2.10.0 debug --default-repo=localhost:5000
Generating tags...
 - mellanox/network-operator -> localhost:5000/mellanox_network-operator:v24.1.0-beta.2-14-gc98f992-dirty
Checking cache...
 - mellanox/network-operator: Not found. Building
Starting build...
Building [mellanox/network-operator]...
Target platforms: [linux/arm64]
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
Cleaning up...
 - No resources found
build [mellanox/network-operator] failed: exit status 1. Docker build ran into internal error. Please retry.
If this keeps happening, please open an issue..
make: *** [dev-skaffold] Error 1
```

Reply:
I wonder what the gap is here - this works for me on a very similar setup - i.e. Mac Silicon + `zsh`.

Reply:
> I think it's related to how this script is started.

What do you think is the gap here?

Reply:
Try exporting a variable in the `hack/scripts/minikube-install.sh` like `eval $(export CUSTOM_VAR="blabla")` and then echo that env variable after calling that script in the make target like:
```
 dev-minikube: $(MINIKUBE) ## Create a minikube cluster for development.
       CLUSTER_NAME=$(MINIKUBE_CLUSTER_NAME) MINIKUBE_BIN=$(MINIKUBE) $(CURDIR)/hack/scripts/minikube-install.sh
       @echo "${CUSTOM_VAR}"
```

I believe that we face this one https://stackoverflow.com/questions/7507810/how-to-source-a-script-in-a-makefile, but you said it's working on your env, so it might not be the case.

Reply:
Thanks for helping me debug this - I've updated the command to use `--detect-minikube=false` and added an inline `eval` to take the correct env variables.